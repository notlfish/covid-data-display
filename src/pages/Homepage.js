import { useState } from 'react';
import CountriesTable from '../components/CountriesTable';
import SearchBar from '../components/SearchBar';
import ToolBar from '../components/ToolBar';

const Homepage = () => {
  const [keyword, setKeyword] = useState('');

  const handleKeywordChange = (e) => {
    setKeyword(e.target.value.toLowerCase());
  };

  return (
    <>
      <header className="bg-pink-odd"><ToolBar /></header>
      <main className="bg-pink-odd">
        <SearchBar handleChange={handleKeywordChange} />
        <CountriesTable keyword={keyword} />
      </main>
    </>
  );
};

export default Homepage;
