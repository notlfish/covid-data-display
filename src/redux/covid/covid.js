import fetchTodayData from '../../utils/stats-api';

const FETCH_SA_DATA = 'covid-data-display/countries/FETCH_SA_DATA';
const ADD_SA_DATA = 'covid-data-display/countries/ADD_SA_DATA';
const API_SUCCESS = 'covid-data-display/countries/API_SUCCESS';
const API_FAILURE = 'covid-data-display/countries/API_FAILURE';

const initialSAData = {
  countries: [],
  loaded: false,
};

export const fetchSAData = () => (dispatch) => {
  dispatch({ type: FETCH_SA_DATA });
  return fetchTodayData().then(
    (countries) => {
      dispatch({ type: API_SUCCESS });
      dispatch({ type: ADD_SA_DATA, countries });
    },
    (error) => {
      dispatch({ type: API_FAILURE, error });
    },
  );
};

const covidSAReducer = (state = initialSAData, action) => {
  switch (action.type) {
    case ADD_SA_DATA:
      return {
        countries: [...state.countries, ...action.countries],
        loaded: true,
      };
    default:
      return state;
  }
};

export default covidSAReducer;
