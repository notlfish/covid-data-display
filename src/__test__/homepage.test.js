import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import store from '../redux/configureStore';
import { fetchSAData } from '../redux/covid/covid';
import Homepage from '../pages/Homepage';

jest.mock('../utils/stats-api');

describe('Homepage snapshot test', () => {
  it('Home page matches snapshot', async () => {
    await store.dispatch(fetchSAData());
    const history = createMemoryHistory();
    history.push('/');
    const homePage = renderer
      .create(<Provider store={store}><Router history={history}><Homepage /></Router></Provider>)
      .toJSON();
    expect(homePage).toMatchSnapshot();
  });
});

describe('Interaction tests', () => {
  it('Displayed countries change when using the search bar', async () => {
    const history = createMemoryHistory();
    history.push('/');
    render(<Provider store={store}><Router history={history}><Homepage /></Router></Provider>);
    const searchBar = screen.getByPlaceholderText('Type to search for a country by name');

    expect(screen.getByText('Argentina')).toBeInTheDocument();
    expect(screen.getByText('Brazil')).toBeInTheDocument();

    fireEvent.change(searchBar, { target: { value: 'Arg' } });

    expect(screen.getByText('Argentina')).toBeInTheDocument();
    expect(screen.queryByText('Brazil')).not.toBeInTheDocument();

    fireEvent.change(searchBar, { target: { value: '' } });

    expect(screen.getByText('Argentina')).toBeInTheDocument();
    expect(screen.getByText('Brazil')).toBeInTheDocument();
  });
});
