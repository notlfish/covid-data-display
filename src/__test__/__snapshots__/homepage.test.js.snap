// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage snapshot test Home page matches snapshot 1`] = `
Array [
  <header
    className="bg-pink-odd"
  >
    <nav
      className="toolbar"
    >
      <a
        aria-current="page"
        className="d-block active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
            fillRule="evenodd"
          />
        </svg>
      </a>
      <h4
        className="capitalize navTitle"
      >
        South America COVID statistics
      </h4>
      <div
        className="rightIcons"
      >
        <svg
          className="pl-10"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <svg
          className="pl-10"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          />
        </svg>
      </div>
    </nav>
  </header>,
  <main
    className="bg-pink-odd"
  >
    <div
      className="searchBarContainer bg-main-pink"
    >
      <div
        className="inputContainer"
      >
        <input
          className="inputField"
          name="searchKeyword"
          onChange={[Function]}
          placeholder="Type to search for a country by name"
          type="text"
          value=""
        />
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M456.69 421.39L362.6 327.3a173.81 173.81 0 0034.84-104.58C397.44 126.38 319.06 48 222.72 48S48 126.38 48 222.72s78.38 174.72 174.72 174.72A173.81 173.81 0 00327.3 362.6l94.09 94.09a25 25 0 0035.3-35.3zM97.92 222.72a124.8 124.8 0 11124.8 124.8 124.95 124.95 0 01-124.8-124.8z"
          />
        </svg>
      </div>
    </div>
    <ul
      className="countriesList bg-pink-even"
    >
      <li>
        <a
          aria-current={null}
          href="/details/argentina"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <svg
              className="float-right"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fillRule="evenodd"
              />
            </svg>
            <img
              alt="Argentina map"
              className="shape"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/samerica/ar/64.png"
            />
            <div
              className="text-right"
            >
              <h3
                className="uppercase"
              >
                Argentina
              </h3>
              <p>
                531293
                 
                open cases
              </p>
            </div>
          </div>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/details/bolivia"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <svg
              className="float-right"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fillRule="evenodd"
              />
            </svg>
            <img
              alt="Bolivia map"
              className="shape"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/samerica/bo/64.png"
            />
            <div
              className="text-right"
            >
              <h3
                className="uppercase"
              >
                Bolivia
              </h3>
              <p>
                77449
                 
                open cases
              </p>
            </div>
          </div>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/details/brazil"
          onClick={[Function]}
        >
          <div
            className="container"
          >
            <svg
              className="float-right"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
                fillRule="evenodd"
              />
            </svg>
            <img
              alt="Brazil map"
              className="shape"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/samerica/br/64.png"
            />
            <div
              className="text-right"
            >
              <h3
                className="uppercase"
              >
                Brazil
              </h3>
              <p>
                2989912
                 
                open cases
              </p>
            </div>
          </div>
        </a>
      </li>
    </ul>
  </main>,
]
`;
